rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -588,83 +602,148 @@ private Integer addFederatorPublicKeyMultikey(boolean dryRun, BtcECKey btcKey, E
     }
 
     /**
-     * Commits the currently pending federation.
-     * That is, the retiring federation is set to be the currently active federation,
-     * the active federation is replaced with a new federation generated from the pending federation,
-     * and the pending federation is wiped out.
-     * Also, UTXOs are moved from active to retiring so that the transfer of funds can
-     * begin.
+     * Commits the currently pending federation
+     * after checking conditions are met to do so.
      * @param dryRun whether to just do a dry run
-     * @param hash the pending federation's hash. This is checked the execution block's pending federation hash for equality.
-     * @return 1 upon success, -1 if there was no pending federation, -2 if the pending federation was incomplete,
-     * -3 if the given hash doesn't match the current pending federation's hash.
+     * @param pendingFederationHash the pending federation's hash. This is checked to match the execution block's pending federation hash.
+     * @return PENDING_FEDERATION_NON_EXISTENT if there was no pending federation,
+     * INSUFFICIENT_MEMBERS if the pending federation was incomplete,
+     * PENDING_FEDERATION_MISMATCHED_HASH if the given hash doesn't match the current pending federation's hash.
+     * SUCCESSFUL upon success.
      */
-    protected Integer commitFederation(boolean dryRun, Keccak256 hash, BridgeEventLogger eventLogger) {
+    protected FederationChangeResponseCode commitFederation(boolean dryRun, Keccak256 pendingFederationHash, BridgeEventLogger eventLogger) {
```

Comment:
Is this protected just to be accessed by the tests or is there another reason?

Reply:
yes, to be accessed by the tests